From 1d80b54aefd1f5f361990b1223c6fe308a8d3085 Mon Sep 17 00:00:00 2001
From: Ricardo Garcia <rgarcia@igalia.com>
Date: Mon, 21 Apr 2025 12:30:05 +0200
Subject: [PATCH] Fix missing #include <cstdint> causing build failures

Tested on Fedora 42 with clang.

Upstream-Status: Submitted [https://github.com/google/amber/pull/1092]

Fixes:
http://errors.yoctoproject.org/Errors/Details/852849/

In file included from TOPDIR/tmp/work/core2-64-oe-linux/vulkan-cts/1.4.1.0/git/external/amber/src/src/type.cc:15:
TOPDIR/tmp/work/core2-64-oe-linux/vulkan-cts/1.4.1.0/git/external/amber/src/src/type.h:56:39: error: 'uint32_t' has not been declared
   56 |   static bool IsInt8(FormatMode mode, uint32_t num_bits) {
      |                                       ^~~~~~~~

Signed-off-by: Martin Jansa <martin.jansa@gmail.com>
---
 src/tokenizer.h | 1 +
 src/type.h      | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/tokenizer.h b/src/tokenizer.h
index 53efec60..ea4f19df 100644
--- a/src/tokenizer.h
+++ b/src/tokenizer.h
@@ -16,6 +16,7 @@
 #define SRC_TOKENIZER_H_
 
 #include <cstdlib>
+#include <cstdint>
 #include <memory>
 #include <string>
 
diff --git a/src/type.h b/src/type.h
index c63779ba..97720f29 100644
--- a/src/type.h
+++ b/src/type.h
@@ -16,6 +16,7 @@
 #define SRC_TYPE_H_
 
 #include <cassert>
+#include <cstdint>
 #include <memory>
 #include <string>
 #include <vector>
